src/main/fc/fc_rc.c

```diff
@@ -179,49 +190,43 @@ static void checkForThrottleErrorResetState(uint16_t rxRefreshRate)
 
 void processRcCommand(void)
 {
-    static float rcCommandInterp[4] = { 0, 0, 0, 0 };
-    static float rcStepSize[4] = { 0, 0, 0, 0 };
-    static int16_t rcInterpolationStepCount;
-    static uint16_t currentRxRefreshRate;
-
-    if (isRXDataNew) {
-        currentRxRefreshRate = constrain(getTaskDeltaTime(TASK_RX),1000,20000);
-        if (isAntiGravityModeActive()) {
-            checkForThrottleErrorResetState(currentRxRefreshRate);
-        }
+    if (skipNextInterpolate && !isRXDataNew) {
+        skipNextInterpolate = false;
+        return;
     }
+    skipNextInterpolate = targetPidLooptime < 62;
 
-    const uint8_t interpolationChannels = rxConfig()->rcInterpolationChannels + 2; //"RP", "RPY", "RPYT"
-    uint16_t rxRefreshRate;
-    uint8_t updatedChannel = 0;
+    int updatedChannel = 0;
+    if (isRXDataNew && isAntiGravityModeActive()) {
+        checkForThrottleErrorResetState();
+    }
 
     if (rxConfig()->rcInterpolation) {
-         // Set RC refresh rate for sampling and channels to filter
-        switch (rxConfig()->rcInterpolation) {
-        case RC_SMOOTHING_AUTO:
-            rxRefreshRate = currentRxRefreshRate + 1000; // Add slight overhead to prevent ramps
-            break;
-        case RC_SMOOTHING_MANUAL:
-            rxRefreshRate = 1000 * rxConfig()->rcInterpolationInterval;
-            break;
-        case RC_SMOOTHING_OFF:
-        case RC_SMOOTHING_DEFAULT:
-        default:
-            rxRefreshRate = rxGetRefreshRate();
-        }
+        if (isRXDataNew) {
+            if (debugMode == DEBUG_RC_INTERPOLATION) {
+                debug[0] = lrintf(rcCommand[0]);
+                debug[1] = lrintf(getTaskDeltaTime(TASK_RX) / 1000);
+            }
+
+             // Set RC refresh rate for sampling and channels to filter
+            switch (rxConfig()->rcInterpolation) {
+                case RC_SMOOTHING_AUTO:
+                    rxRefreshRate = currentRxRefreshRate + 1000; // Add slight overhead to prevent ramps
+                    break;
+                case RC_SMOOTHING_MANUAL:
+                    rxRefreshRate = 1000 * rxConfig()->rcInterpolationInterval;
+                    break;
+                case RC_SMOOTHING_OFF:
+                case RC_SMOOTHING_DEFAULT:
+                default:
+                    rxRefreshRate = rxGetRefreshRate();
+            }
 
-        if (isRXDataNew && rxRefreshRate > 0) {
             rcInterpolationStepCount = rxRefreshRate / targetPidLooptime;
```

Comment:
targetPidLooptime should have an inverse recalculated somewhere.

Reply:
as far as I can tell it's a float that is 31.25, 62.5, 125 etc..